### PR TITLE
[V3 Audio] Harden Lavalink boot sequence

### PR DESCRIPTION
**Pull request by tekulvw**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
This PR does a lot...
* Ensures that java 1.8+ is installed
  * If 1.8, ensures a compatibility flag is passed to the lavalink launch command. This should fix some lingering issues with lavalink boot looping e.g. #1462.
* Downloads the lavalink jar from our releases page (sorry)
  * This has to happen so we can keep the lavalink version static to ensure compatibility.
* Completely redoes version handling so we only have to modify it in one place `redbot/core/__init__.py`

### CODE REVIEW

File: redbot/core/__init__.py [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+        self.serial = serial
+
+    def __lt__(self, other):
+        my_index = self._levels.index(self.releaselevel)
+        other_index = self._levels.index(other.releaselevel)
+        return self.major < other.major or \
+            self.minor < other.minor or \
+            self.micro < other.micro or \
+            my_index < other_index or \
+            self.serial < other.serial
```
Comment (mikeshardmind) [inline redbot/core/__init__.py]: This should compare as 
```py
return (self.major, self.minor, self.micro, my_index, self.serial) < (other.major, other.minor, other.micro, other_index, other.serial)
```
The way you have it here with nested or would have 3.0.1 be < 3.1.0

**Review by mikeshardmind — COMMENTED** [review]
Seems to have an issue with version info comparison

File: redbot/core/__init__.py (hunk 1 above)
Comment (tekulvw) [reply redbot/core/__init__.py]: 3.0.1 is less than 3.1.0?

Comment (mikeshardmind) [reply redbot/core/__init__.py]: my mistake, I meant that both of these version comparisons evaluate true by the above:

3.0.1 < 3.1.0
3.1.0 < 3.0.1

**Review by palmtree5 — APPROVED** [review]

**Pull request merged**